core/src/modules/ops/record.ts

```diff
@@ -831,12 +831,16 @@ export namespace RecordOps {
     }
 
     const recordProperties = await RecordProperty.findAll({
-      where: { unique: true, rawValue: rawValues },
+      where: {
+        propertyId: uniqueProperties.map((p) => p.id),
+        unique: true,
+        rawValue: rawValues,
+      },
       include: [GrouparooRecord],
     });
 
     const properties = await PropertiesCache.findAllWithCache(
-      undefined,
+      source instanceof Source ? source.modelId : undefined,
       "ready"
     );
 
```

Comment:
Could the same thing happen within the same model with two unique properties?

Say we have two unique id properties, user_id and external_id that are integers.

Record 1: user_id = 1, external_id=2
Record 2: user_id = 2, external_id = 3

I _think_ when matching this would look for any record property that has the value 2 or the value 3. Since we have a unique property (external_id) with the value 2, it would match that first record when it shouldn't.

Should the loop below that does the matching only match on recordProperties for a specific key?